packages/core/content-manager/server/controllers/relations.js

```diff
@@ -26,57 +26,52 @@ module.exports = {
 
     // idsToOmit: used to exclude relations that the front already added but that were not saved yet
     // idsToInclude: used to include relations that the front removed but not saved yes
-    const { component, entityId, idsToOmit, idsToInclude, _q, ...query } = ctx.request.query;
+    const { entityId, idsToOmit, idsToInclude, _q, ...query } = ctx.request.query;
 
-    const sourceModelUid = component || model;
-
-    const sourceModel = strapi.getModel(sourceModelUid);
-    if (!sourceModel) {
+    const modelSchema = strapi.getModel(model);
+    if (!modelSchema) {
       return ctx.badRequest("The model doesn't exist");
     }
 
-    const permissionChecker = getService('permission-checker').create({
-      userAbility,
-      model,
-    });
-
-    if (permissionChecker.cannot.read()) {
-      return ctx.forbidden();
-    }
-
-    const attribute = sourceModel.attributes[targetField];
+    const attribute = modelSchema.attributes[targetField];
     if (!attribute || attribute.type !== 'relation') {
       return ctx.badRequest("This relational field doesn't exist");
     }
 
-    // TODO: find a way to check field permission for component
-    if (!component && permissionChecker.cannot.read(null, targetField)) {
-      return ctx.forbidden();
-    }
-
-    if (entityId) {
-      const entityManager = getService('entity-manager');
+    const isComponent = modelSchema.modelType === 'component';
 
-      const entity = await entityManager.findOneWithCreatorRoles(entityId, model);
-
-      if (!entity) {
-        return ctx.notFound();
-      }
+    // RBAC checks when it's a content-type
+    // TODO: do RBAC check for components too
+    if (!isComponent) {
+      const permissionChecker = getService('permission-checker').create({
+        userAbility,
+        model,
+      });
 
-      if (!component && permissionChecker.cannot.read(entity, targetField)) {
+      if (permissionChecker.cannot.read(null, targetField)) {
         return ctx.forbidden();
       }
-      // TODO: find a way to check field permission for component
-      if (component && permissionChecker.cannot.read(entity)) {
-        return ctx.forbidden();
+
+      if (entityId) {
```

Comment:
shouldn't the entityID be mandatory if it is not a component? The user could make the same request without the entity ID and

Reply:
Also, I can not edit role permissions,I get a Validation error 
"Permissions have to be defined all together for a subject field or not at all"

Am I missing some step?

Reply:
If we make it mandatory, we could merge this rbac logic from both endpoints, something like `checkRBACPermissions`

Reply:
When you create an entry, you need to use this route and you have no entityId to provide. That's why it's optional :)

Regarding the permissions error you have, I think you have an issue with your license/environment. We can check that !